--- a/os_dep/linux/os_intfs.c	2024-05-06 23:23:58.000000000 +0800
+++ b/os_dep/linux/os_intfs.c	2024-11-05 04:13:55.891780209 +0800
@@ -617,11 +617,7 @@
 	u8 rtnl_lock_needed = rtw_rtnl_lock_needed(dvobj);
 
 #ifdef CONFIG_RTW_NAPI
-#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 1,0)
-	netif_napi_add(ndev, &adapter->napi, rtw_recv_napi_poll, RTL_NAPI_WEIGHT);
-#else
 	netif_napi_add(ndev, &adapter->napi, rtw_recv_napi_poll);
-#endif
 #endif /* CONFIG_RTW_NAPI */
 
 #if defined(CONFIG_IOCTL_CFG80211)
